cmd/zetatool/cli/tss_balances.go

```diff
@@ -0,0 +1,506 @@
+
+The network argument must be one of: mainnet, testnet, localnet, devnet
+
+Example:
+  zetatool tss-balances mainnet
+  zetatool tss-balances testnet
+  zetatool tss-balances localnet --config custom_config.json`,
+		Args: cobra.ExactArgs(1),
+		RunE: getTSSBalances,
+	}
+}
+
+func getTSSBalances(cmd *cobra.Command, args []string) error {
+	network := args[0]
+
+	// Use custom rpc config if provided
+	configFile, err := cmd.Flags().GetString(config.FlagConfig)
+	if err != nil {
+		return fmt.Errorf("failed to read value for flag %s: %w", config.FlagConfig, err)
+	}
+
+	cfg, err := config.GetConfigByNetwork(network, configFile)
+	if err != nil {
+		return fmt.Errorf("failed to get config: %w", err)
+	}
+
+	if cfg.ZetaChainRPC == "" {
+		return fmt.Errorf("ZetaChainRPC is not configured for network %s", network)
+	}
+
+	// Create ZetaCore client and fetch TSS
+	zetacoreClient, err := rpc.NewCometBFTClients(cfg.ZetaChainRPC)
+	if err != nil {
+		return fmt.Errorf("failed to create zetacore client: %w", err)
+	}
+
+	ctx := context.Background()
+
+	tssHistoryRes, err := zetacoreClient.Observer.TssHistory(ctx, &observertypes.QueryTssHistoryRequest{})
+	if err != nil {
+		return fmt.Errorf("failed to fetch TSS history: %w", err)
+	}
+
+	if len(tssHistoryRes.TssList) == 0 {
+		return fmt.Errorf("no TSS entries found")
+	}
+
+	for i, tss := range tssHistoryRes.TssList {
+		if i > 0 {
+			fmt.Println() // Add spacing between TSS entries
+		}
+		fmt.Printf("=== TSS %d of %d ===\n", i+1, len(tssHistoryRes.TssList))
+
+		if err := printTSSBalances(ctx, cfg, tss, network, zetacoreClient.Observer); err != nil {
+			fmt.Printf("Error fetching balances for TSS (height %d): %v\n", tss.FinalizedZetaHeight, err)
+			continue
+		}
+	}
+
+	return nil
+}
+
+// getSymbolForChain returns the native token symbol for a chain
+func getSymbolForChain(chain pkgchains.Chain) string {
+	if symbol, ok := networkSymbols[chain.Network]; ok {
+		return symbol
+	}
+	return ""
+}
+
+// getRPCForChain returns the RPC URL for a given chain from config
+func getRPCForChain(cfg *config.Config, chain pkgchains.Chain) string {
+	switch chain.Network {
+	case pkgchains.Network_eth:
+		return cfg.EthereumRPC
+	case pkgchains.Network_bsc:
+		return cfg.BscRPC
+	case pkgchains.Network_polygon:
+		return cfg.PolygonRPC
+	case pkgchains.Network_base:
+		return cfg.BaseRPC
+	case pkgchains.Network_arbitrum:
+		return cfg.ArbitrumRPC
+	case pkgchains.Network_optimism:
+		return cfg.OptimismRPC
+	case pkgchains.Network_avalanche:
+		return cfg.AvalancheRPC
+	case pkgchains.Network_worldchain:
+		return cfg.WorldRPC
+	case pkgchains.Network_solana:
+		return cfg.SolanaRPC
+	case pkgchains.Network_ton:
+		return cfg.TonRPC
+	case pkgchains.Network_sui:
+		return cfg.SuiRPC
+	default:
+		return ""
+	}
+}
+
+// printTSSBalances fetches and prints TSS address balances across all chains
+func printTSSBalances(
```

Comment:
It would be nice to break down the contents of this this function into smaller, per-chain functions? For readability ... maybe even a separate file?